extensions/common/azure/azure-blob-core/src/main/java/org/eclipse/edc/azure/blob/BlobStoreCoreExtension.java

```diff
@@ -42,10 +43,11 @@ public String name() {
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var blobstoreEndpointTemplate = context
-                .getSetting(EDC_BLOBSTORE_ENDPOINT_TEMPLATE, "https://%s.blob.core.windows.net");
-
-        var blobStoreApi = new BlobStoreApiImpl(vault, blobstoreEndpointTemplate);
+        var blobStoreApi = new BlobStoreApiImpl(vault,
+                blobStoreCoreExtensionConfig.blobstoreEndpointTemplate(),
+                blobStoreCoreExtensionConfig.blockSize(),
+                blobStoreCoreExtensionConfig.maxConcurrency(),
+                blobStoreCoreExtensionConfig.maxSingleUploadSize());
```

Comment:
please pass the configuration all the way down to `DefaultBlobAdapter`, it will make everything more readable

Reply:
Good point, changed in [here](https://github.com/eclipse-edc/Technology-Azure/pull/298/commits/a20a5bd83a4033c9249b24ca810cbacdc8a248df).